src/main/java/com/hz6826/clockin/init/DatabaseInit.java

```diff
@@ -3,52 +3,192 @@
+        DatabaseMetaData metaData = null;
+        try {
+            metaData = DBconnection.getMetaData();
+        } catch (SQLException e) {
+            throw new RuntimeException("get database metadata error");
+        }
 
-    public DatabaseInit(){
-        //读取配置文件
-        Gson gson = new Gson();
+        String dataBaseName = "";
+        try {
+            dataBaseName = metaData.getDatabaseProductName();
+        } catch (SQLException e) {
+            throw new RuntimeException("get database type error");
+        }
+
+
+        JsonArray sql_list = JsonParser.parseString(loadSource("sql/index.sql")).getAsJsonObject().getAsJsonArray();
+
+        if (sql_list.getAsJsonArray().isEmpty()) {
+            throw new RuntimeException("no sql list");
+        }
 
+        boolean databaseExists = false;
 
+        for (JsonElement databaseElm : sql_list) {
+            if (databaseElm.getAsString().equalsIgnoreCase(dataBaseName)) {
+                databaseExists = true;
+                break;
+            }
+        }
 
-        //获取当前数据库的类型
+        if (!databaseExists) {
+            throw new RuntimeException("no sql list");
+        }
 
         //查找是否有匹配的数据库文件
 
+        JsonObject sqlJson = JsonParser.parseString(loadSource("sql" + dataBaseName + "/index.json")).getAsJsonObject().getAsJsonObject();
+        if (!sqlJson.get("sql_name").getAsString().equalsIgnoreCase(dataBaseName)) {
+            throw new RuntimeException("sql name not match,may be the sql is bad");
+        }
+        ;
+
+        JsonObject table_sql_list_array = sqlJson.get("tables").getAsJsonObject();
+
+        this.sql = new sql_list(dataBaseName, table_sql_list_array);
+
         //有则添加
 
         //没有报错
     }
 
-    public boolean createTable(String name){
 
+    public boolean export_data(String path) {
+
+        return true;
+    }
+
+    public boolean import_data(String path) {
+        return true;
+    }
+
+    public boolean createTable(String name) {
+
+        String sql=this.sql.sql_tables.get(name).getAsString();
+        if (sql == null) {
+            return false;
+        }
+        try (Statement statement = this.connection.createStatement()) {
+            statement.executeUpdate(sql);
+
+        } catch (SQLException e) {
+           throw  new RuntimeException("create table error",e);
+        }
         return true;
     }
+
+
     public boolean deleteTable(String name) {
+        
         return true;
     }
-    public boolean createAll(String name){
+
+    public boolean createAll(String name) {
+        try{
+            this.connection.setAutoCommit(false);
+
+        }catch (SQLException e){
+            throw  new RuntimeException("can not create Transaction",e);
+        }
+
+        JsonObject jsonObject = this.sql.sql_tables;
+
+        // 获取所有键值对
+        Set<Map.Entry<String, com.google.gson.JsonElement>> entries = jsonObject.entrySet();
+
+        // 遍历并打印所有数据
+
+        String currentKey="";
+        try {
```

Comment:
```suggestion
        try (Connection conn =  ...) {
```
try with resource会自动处理的，而且用PreparedStatement更好